whois-commons/src/main/resources/nrtm_schema.sql

```diff
@@ -19,58 +35,72 @@ create table `source`
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
-DROP TABLE IF EXISTS `version_information`;
-create table `version_information`
+CREATE TABLE `version_info`
 (
     `id`             int unsigned NOT NULL AUTO_INCREMENT,
     `source_id`      int unsigned NOT NULL,
-    `version`        int unsigned NOT NULL DEFAULT '0',
-    `session_id`     varchar(128) NOT NULL DEFAULT '',
-    `type`           varchar(128) NOT NULL DEFAULT '',
-    `last_serial_id` int          NOT NULL DEFAULT '0',
+    `version`        int unsigned NOT NULL,
+    `session_id`     varchar(128) NOT NULL,
+    `type`           varchar(128) NOT NULL,
+    `last_serial_id` int          NOT NULL,
     PRIMARY KEY (`id`),
-    CONSTRAINT `version_information__source_fk` FOREIGN KEY (`source_id`) REFERENCES `source` (`id`),
-    UNIQUE KEY `version_information__source__version_uk` (`source_id`, `version`),
-    UNIQUE KEY `version_information__session__version_uk` (`session_id`, `version`),
-    UNIQUE KEY `version_information__type__last_serial_id_uk` (`type`, `last_serial_id`)
+    CONSTRAINT `version__source__fk` FOREIGN KEY (`source_id`) REFERENCES `source` (`id`),
+    UNIQUE KEY `version__source__version__uk` (`source_id`, `version`),
+    UNIQUE KEY `version__session__version__uk` (`session_id`, `version`),
+    UNIQUE KEY `version__type__last_serial_id__uk` (`type`, `last_serial_id`)
 ) ENGINE = InnoDB
   DEFAULT CHARSET = utf8;
 
-DROP TABLE IF EXISTS `snapshot_file`;
-create table `snapshot_file`
+CREATE TABLE `snapshot_object`
 (
-    `id`         int(10) unsigned NOT NULL AUTO_INCREMENT,
-    `version_id` int(10) unsigned NOT NULL DEFAULT '0',
-    `name`       varchar(256)     NOT NULL DEFAULT '',
-    `hash`       varchar(128)     NOT NULL DEFAULT '',
-    `created`    bigint unsigned  not null default unix_timestamp(),
+    `id`          int unsigned NOT NULL AUTO_INCREMENT,
+    `version_id`  int unsigned NOT NULL,
+    `object_id`   int unsigned NOT NULL,
+    `sequence_id` int unsigned NOT NULL,
+    `rpsl`        longtext     NOT NULL,
```

Comment:
Are there any downsides to using `LONGTEXT` versus `LONGBLOB`? is there any implicit character conversion going on? Does it need to be `LONGTEXT` in order to use json functions on it?

Reply:
longtext will do translation to the table charset, UTF8 in this case. i thought that would be a good idea since we'd be serving UTF8 JSON, i.e. we shouldn't put anything but UTF8 in the tables or we won't serve what the HTTP charset is saying we are. thinking a different way about it i'm not so sure -- let's talk about it.

Reply:
As discussed LONGTEXT is OK